domains/client/domain-operator/src/tests.rs

```diff
@@ -877,48 +862,203 @@ async fn test_bad_invalid_state_transition_proof_is_rejected() {
     // trace index 1 is the index of the state root after applying inherent timestamp extrinsic
     // trace index 2 is the index of the state root after applying above balance transfer extrinsic
     // trace index 3 is the index of the state root after applying BlockBuilder::finalize_block
-    for mismatch_index in 0..valid_receipt.execution_trace.len() {
-        let fraud_proof = fraud_proof_generator
-            .generate_invalid_state_transition_proof(
-                GENESIS_DOMAIN_ID,
-                mismatch_index.try_into().expect(
-                    "execution trace index should always be convertible into u32 from usize; qed",
-                ),
-                &valid_receipt,
-                valid_receipt_hash,
-            )
-            .unwrap();
+    for trace_diff_type in [
+        TraceDiffType::Shorter,
+        TraceDiffType::Longer,
+        TraceDiffType::Mismatch,
+    ] {
+        for mismatch_index in 0..valid_receipt.execution_trace.len() {
+            let dummy_execution_trace = match trace_diff_type {
+                TraceDiffType::Shorter => valid_receipt
+                    .execution_trace
+                    .clone()
+                    .drain(..)
+                    .take(mismatch_index + 1)
+                    .collect(),
+                TraceDiffType::Longer => {
+                    let mut long_trace = valid_receipt.execution_trace.clone();
+                    long_trace.push(H256::default());
+                    long_trace
+                }
+                TraceDiffType::Mismatch => {
+                    let mut modified_trace = valid_receipt.execution_trace.clone();
+                    modified_trace[mismatch_index] = H256::default();
+                    modified_trace
+                }
+            };
+
+            let result_execution_phase = fraud_proof_generator.find_mismatched_execution_phase(
+                valid_receipt.domain_block_hash,
+                &valid_receipt.execution_trace,
+                &dummy_execution_trace,
+            );
+
+            if result_execution_phase.is_err()
+                || result_execution_phase
+                    .as_ref()
+                    .is_ok_and(|maybe_execution_phase| maybe_execution_phase.is_none())
+            {
+                continue;
+            }
```

Comment:
IIUC `Error` or `Ok(None)` should be unexpected, we should panic in this case, right?